config/config_params.go

```diff
@@ -335,6 +335,9 @@ type Config struct {
 	// Configures MTU auto-detection.
 	MTUIfacePattern *regexp.Regexp `config:"regexp;^((en|wl|ww|sl|ib)[opsx].*|(eth|wlan|wwan).*)"`
 
+	// Provides information on whether Felix should maintain endpoints in the absence of Calico CNI
+	InterfaceConfigurationEnabled bool `config:"bool;true"`
```

Comment:
In fact, are privs needed to **read** the various settings that we are interested in?  Perhaps our approach should be to read first and then no-op if no change needed.

Reply:
In fact, are privs needed to **read** the various settings that we are interested in?  Perhaps our approach should be to read first and then no-op if no change needed.

Reply:
> Could the CNI plugin instead leave a marker, like a file in the pod filesystem, and then Felix checks for that?
I think this works in terms of telling whether the Calico CNI is running or not. I originally thought of controlling this through a configuration parameter because the other thing that affects whether this will work or not is whether or not node is being run in a privileged container or not. Since that's something that has to be changed on the k8s level, I figured it would match up better with also setting extra configuration but ideally we would be able to switch this on or off based on whether the container is privileged **and** the Calico CNI is running or not.

Reply:
>In fact, are privs needed to read the various settings that we are interested in? Perhaps our approach should be to read first and then no-op if no change needed.

My understanding was that we set these values to change how the kernel behaves so I don't think that any of the code we write specifically needs to read out the values. I would need to test if we could `os.Stat` the files in an unprivileged container in order to determine whether or not we really need to set the values for the host, but that might lead to some weird scenarios where we have the ability to read the files in an unprivileged container but then when we notice that they're not available, fail to create them because of the permissions.